src/rpc/mining.cpp

```diff
@@ -447,6 +447,19 @@ UniValue getblocktemplate(const JSONRPCRequest& request)
     if (IsInitialBlockDownload())
         throw JSONRPCError(RPC_CLIENT_IN_INITIAL_DOWNLOAD, "Bitcoin is downloading blocks...");
 
+    if (!lpval.isNull()) {
```

Comment:
In commit "Replace csBestBlock/cvBlockChanged with validationinterface"

Is the `!` here a mistake? I could see how the RPC call might return too early if lpval is null, but if it's non-null and actually set to a hash, it seems like this would only lead to unnecessary delays.